### PR TITLE
Exports Download Mocha Tests

### PR DESCRIPTION
**Pull request by biyeun**
more mocha tests for the more complex part of the exports download the `downloadExportFormController`. What's not present are the pending tests for `DownloadProgressController`

@benrudolph @czue

### CODE REVIEW

File: corehq/apps/export/static/export/spec/exportDownloadService.polling.spec.mo.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        it('registers a reset', function () {
+            assert.equal(DnldExpData.exportDownloadService.downloadError, 'default');
+            assert.equal(DnldExpData.exportDownloadService._numErrors, 5);
+            DnldExpData.exportDownloadService.resetDownload();
+            DnldExpData.currentScope.$apply();
+            assert.isNull(DnldExpData.exportDownloadService.downloadError);
+            assert.equal(DnldExpData.exportDownloadService._numErrors, 0);
+        });
+    });
+});
```
Comment (benrudolph) [inline corehq/apps/export/static/export/spec/exportDownloadService.polling.spec.mo.js]: nice work on these. error handling is annoying but separates a professional product from slop

**Comment by benrudolph** [comment]
:heart_decoration: :heart_decoration: 
:+1: once tests passed (travis looks to be way behind)

**Comment by benrudolph** [comment]
actually js tests passed and that's what matters for this PR. travis is broken at the moment

**Comment by czue** [comment]
awesome! so much :white_check_mark: to :tada:

**Pull request merged**